packages/remix-dev/vite/plugin.ts

```diff
@@ -446,6 +446,13 @@ export const remixVitePlugin: RemixVitePlugin = (options = {}) => {
 
         viteChildCompiler = await createViteDevServer({
           ...viteUserConfig,
+          server: {
+            ...viteUserConfig.server,
+            // when parent compiler runs in middleware mode to support
+            // custom servers, we don't want the child compiler also
+            // run in middleware mode as that will cause websocket port conflicts
+            middlewareMode: false,
+          },
```

Comment:
@markdalgleish maybe we just want to unset all `server` options for child compiler since we don't want to use it as a server at all?